File: 2024/soutez/04-Horolezec/solution/solution.py
import math


if __name__ == "__main__":
    n = int(input())

    x = 0
    y = 0

    max_height = y
    
    for i in range(n):
        u, l = map(int, input().split(" "))

        u_r = math.radians(u)

        delta = (math.cos(u_r)*l, math.sin(u_r)*l)

        x += delta[0]
        y += delta[1]

        if y > max_height: max_height = y

    
    cutoff = lambda n : int(n*100)/100

    print(cutoff(max_height))
    print(cutoff(x))

File: 2024/soutez/04-Horolezec/solution/solution.cs
using System;

class Program
{
    static void Main()
    {
        double maxHeight=0; // nejvyssi bod
        double distance=0; // celkova delka
        double currentHeight=0; // aktualni vyska
        
        int n = int.Parse(Console.ReadLine());
        
        for (int i=0; i<n; i++)
        {
            string[] input = Console.ReadLine().Split(' ');
            double angle = int.Parse(input[0])*(Math.PI/180);
            double lenght = int.Parse(input[1]);
            
            distance += lenght*Math.Cos(angle);
            
            currentHeight += lenght*Math.Sin(angle);
            if (currentHeight>maxHeight)
            {
                maxHeight = currentHeight;
            }
        }
        
        Console.Write($"{Math.Round(maxHeight,2)}\n{Math.Round(distance,2)}");
        
    }
}